interface/src/app/volcano-plot/volcano-plot.js

```diff
@@ -4,50 +4,134 @@
  * via plotData.
  */
 angular.module('adage.volcano-plot', [
+  'ngResource',
   'ngVega',
   'greenelab.stats',
+  'adage.utils',
   'adage.volcano-plot-vgspec'
 ])
 
-.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec',
-  function VolcanoPlotCtrl($scope, VolcanoPlotSpec) {
-    // define controller instance variables that link with vega
-    this.spec = VolcanoPlotSpec;
-    this.data = $scope.plotData;
-    // FIXME: can't get selectedNodes out this way... need to use view api?
-    // $scope.plotData.selectedNodes: $scope.selectedNodes
+// TODO when refactoring for issue #76, this service belongs in the top-level
+//      src/app/ module called `node` in its `services.js` file
+.factory('NodeService', ['$resource', 'ApiBasePath',
+  function($resource, ApiBasePath) {
+    // private methods for NodeService
+    var Participation = $resource(ApiBasePath + 'participation/');
+
+    // public methods for NodeService
+    return {
+      getGenesForNodesPromise: function getGenesForNodesPromise(nodes) {
```

Comment:
Good question. Same as with the other named function in ``volcano-plot/view.js ``, I have a hard time deciding if I should name these functions or not.